src/NuGet.Commands/RestoreCommand.cs

```diff
@@ -400,7 +400,7 @@ private bool ResolutionSucceeded(IEnumerable<RestoreTargetGraph> graphs)
         {
             // Get the project graph
             var projectFrameworks = project.TargetFrameworks.Select(f => f.FrameworkName).ToList();
-            if (projectFrameworks.Count > 1)
+            if (projectFrameworks.Count > 1 || targetGraphs.Count() == 0)
```

Comment:
You don't need to enumerate the whole list here, you can use: !targetGraphs.Any()